app/src/main/java/ai/elimu/content_provider/util/FileHelper.kt

```diff
@@ -20,9 +20,9 @@ object FileHelper {
             return null
         }
         val imagesDirectory = context.getExternalFilesDir(Environment.DIRECTORY_PICTURES)
+        val fileName = imageGson.fileUrl.substringAfterLast('/').substringBeforeLast('.')
```

Comment:
@tuancoltech How about this?:
```suggestion
        val fileName = imageGson.checksumMd5
```